src/components/FilterBarInline/FilterBarInline.stories.tsx

```diff
@@ -0,0 +1,1204 @@
+      {
+        id: "history",
+        name: "History",
+        children: [
+          { id: "african-american", name: "African American History" },
+          { id: "united-states", name: "United States History" },
+          { id: "world", name: "World History" },
+        ],
+      },
+    ],
+  },
+];
+const shortMultiSelectItems = [...multiSelectItems];
+shortMultiSelectItems.pop();
+
+const meta: Meta<typeof FilterBarInline> = {
+  title: "Components/Form Elements/FilterBarInline",
+  component: FilterBarInline,
+  argTypes: {
+    className: { control: false },
+    id: { control: false },
+    heading: { control: false },
+    layout: {
+      control: { type: "radio" },
+      options: layoutTypesArray,
+      table: { defaultValue: { summary: "row" } },
+    },
+    onClear: {
+      description:
+        "Function to clear all selectedItems - available through hook",
+    },
+    onSubmit: {
+      description:
+        "Function to handle submit of all selectedItems - needs to be provided from consuming app",
+    },
+    selectedItems: { control: false },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof FilterBarInline>;
+
+/**
+ * Main Story for the FilterBarInline component. This must contains the `args`
+ * and `parameters` properties in this object.
+ */
+export const WithControls: Story = {
+  args: {
+    id: "filterbar-inline-id",
+    heading: "FilterBarInline",
+    layout: "row",
+  },
+  render: (args) => <FilterBarStory {...args} />,
+  parameters: {
+    design: {
+      type: "figma",
+      url: "https://www.figma.com/file/qShodlfNCJHb8n03IFyApM/Main?node-id=87515-95820",
+    },
+    jest: ["FilterBarInline.test.tsx"],
+  },
+};
+
+// The following are additional FilterBar example Stories.
+export const LayoutPatterns: Story = {
+  render: () => <FilterBarLayoutStory />,
+};
+export const UIContainers: Story = {
+  render: () => <FilterBarRowContainerStory />,
+  name: "UI Containers for Row Layout",
+};
+export const ColumnLayout: Story = {
+  render: () => <FilterBarColumnContainerStory />,
+  name: "UI Containers for Column Layout",
+};
+export const MultipleChildren: Story = {
+  render: () => <FilterBarChildrenStory />,
+};
+
+const FilterBarStory = (args) => {
+  const { onChange, onMixedStateChange, selectedItems, onClear, onClearAll } =
+    useMultiSelect();
+  const selectedFilterItems = [selectedItems];
+  return (
+    <FilterBarInline
+      id={args.id}
+      heading={args.heading}
+      layout={args.layout}
+      onClear={onClearAll}
+      onSubmit={() => console.log(selectedItems)}
+      selectedItems={selectedFilterItems}
+      renderChildren={({ layout, width }) => {
+        return (
+          <MultiSelectGroup
+            id="multiselect-group"
+            labelText="MultiSelect Group"
+            layout={layout}
+            multiSelectWidth={width}
+            renderMultiSelect={({ isBlockElement, multiSelectWidth }) => {
+              return (
```

Comment:
Not a problem but just funny you don't follow the recommendation you gave about splitting this up into smaller functions!